### ISSUE TITLE
Consider move to libadwaita

### ISSUE THREAD

**Issue by MossyMossMoss** [lutris/lutris#3886]
With all core GNOME apps, and many other GTK apps moving to libadwaita ahead of the release of GNOME 42, the lutris client should consider moving to it as well. It will ensure the app does not look out of place with the rest of a user's system and provides many additional features.

**Comment by tannisroot** [comment]
This is definitely a good idea but I don't think we will be able to do this for a while due to needing to support LTS distributions. AFAIK even the latest LTS Ubuntu version doesn't have support for libadwaits/gtk4.

**Comment by A6GibKm** [comment]
Note that libadwaita has not even released yet.

**Comment by mkpaz** [comment]
libadwaita separates Gnome from GTK and right now it looks absolutely [awful](https://www.reddit.com/r/kde/comments/ry20h7/are_there_any_plans_by_kde_to_try_and_tame_the/) on KDE and other Qt based distros. Any project considering to switch from GTK3 to the libadwaita basically should decide whether it's GTK based project for all Linux users or Gnome-based project for Gnome users only.

**Comment by jannuary** [comment]
> and right now it looks absolutely awful on KDE and other Qt based distros

Oh no! Libadwaita apps can no longer get awful CSS patches that break the app, what a travesty!

Either way - it's definitely a question of scope though - Lutris could be more or less platform-neutral, or [explicitly target GNOME](https://gitlab.gnome.org/Teams/Circle/-/issues/31), follow its HIG and use its platform libraries.

**Comment by A6GibKm** [comment]
The resolution of that conversation was that Lutris supports distributions with *slower cadence*, meaning that it is not possible to port to libadwaita, since that would require porting to GTK 4 and not all distributions had it in their repositories it at the time (It wasn't in Debian Sid's at the time).

It looks like GTK 4.4 is part of Debian 12 now though and this could be revisited once libadwaita hits its repository.

**Comment by BrainBlasted** [comment]
I'm definitely still interested in working on this, though my impression was indeed that it should probably wait until libadwaita was shipped on more distros.

**Comment by mossymossmoss** [comment]
Great! It will be good to return to this issue once that is the case

**Comment by HarryMichal** [comment]
This could possibly be split into two separate issues. One for a port to GTK4 which probably require a significant amount of work and the other for the adoption of libadwaita. The adoption of libadwaita will require GTK4 nonetheless and "pure" GTK4 has a lower entry barrier for distributions.

**Comment by appetrosyan** [comment]
Porting to GTK4 might be a necessity at some point. GTK 3 has a few major technical drawbacks that will put Lutris at a disadvantage. 

Libadwaita I'm personally firmly in the camp of not adopting. I'm using Lutris on KDE. It looks badly out of place, but at the very least it respects the system colour palette. If lutris decides to move to Libadwaita, I will personally fork it to Kirigami.

**Comment by appetrosyan** [comment]
> Oh no! Libadwaita apps can no longer get awful CSS patches that break the app, what a travesty!

I think what you meant to say, was

> Oh no, the judgmental buffoons in charge of Gnome can't enforce their subjective (and most often wrong) opinions on the software community at large, further locking down the GTK ecosystem and pushing people towards Qt as the only other alternative. 

What a travesty indeed.

**Comment by TohurTV** [comment]
for the love of god if this happens please make a QT port so at least it looks right in every other DE. not everyone uses GNOME (Apple of Linux) and chooses not to have their opinion shoved down our throats forcing their way onto us. Linux is about freedom and starting to thing GNOME is the fascists of the Linux community

**Comment by jannuary** [comment]
> > Oh no, the … in charge of Gnome can't enforce their subjective (and most often wrong) opinions on the software community at large, further locking down the GTK ecosystem and pushing people towards Qt as the only sane toolkit. 
> 
> What a travesty indeed. 

You're free not to use it. Either way, this is really not a place for arguing about how GNOME is evil or whatever.

**Comment by appetrosyan** [comment]
> You're free not to use it. 

Or to fork it. 

> Either way, this is really not a place for arguing about how GNOME is evil or whatever.

Thankfully, this is exactly not what we're doing. We're discussing the technical drawbacks of adopting `libadwaita`. Its technical drawbacks have a non-zero overlap with Gnome criticism, namely a one-size-fits-all approach.

**Comment by A6GibKm** [comment]
Please let us keep the discussion factual. I would like to take this opportunity to mention that libadwaita can be themed without any inconvenience by setting GTK_THEME.

If a libadwaita app looks out of place in a certain theme or environment, the are two possibilities: The theme has to be updated to support libadwaita and/or GTK 4, or the enviroment needs to set GTK_THEME.

**Comment by tannisroot** [comment]
Anyone who starts a Gnome shit fight afterwards is going to be permanently banned from participating in the repository.

**Comment by BrainBlasted** [comment]
Another option instead of waiting for distros would be making flatpak the first-class distribution method, but that would require getting all runners working in flatpak. Would that be something the Lutris team is interested in?

**Comment by nvidiaLinuxUser** [comment]
+, Lutris looks ugly compared to other GNOME 42 apps.

**Comment by BrainBlasted** [comment]
Lutris now has a release on flathub proper - are all the runners in working order now?

**Comment by axel358** [comment]
I believe this kind of apps should remain neutral, like libreoffice or inkscape do. Using libadwaita is indeed embracing the GNOME HIG. If a gtk3/4 app looks out of place in GNOME just use the adw-gtk3 theme.

**Comment by mossymossmoss** [comment]
It makes sense for Libreoffice and Inkscape to stick with GTK alone, as they are both have fairly complicated GUI and must support Windows, MacOS and Linux. Lutris has very similar UI to most GNOME apps, and is Linux-exclusive, meaning that unlike the aforementioned programs, the switch should be fairly straightforward. Using adw-gtk3 instead, however, will mean that Lutris misses out on all the new features and widgets that Libadwaita offers

**Comment by axel358** [comment]
Linux is not GNOME alone, is not even GNOME and KDE, a fact that those using the above like to ignore. But i guess is alright, if that's what the devs want is a decision I have to respect, just wanted to share my thoughts on the subject

**Comment by cameronbosch** [comment]
I use Lutris occasionally, and even as somebody who prefers KDE Plasma as their primary DE, I think Lutris should move to Libadwaita so that they can take advantage of all of the benefits of it. And to of course not look outdated imo so that more users can enjoy Lutris (that's one of the reasons but not the only reason I mostly moved to Bottles for certain applications).

TL;DR, I would love to see a libadwaita version so that more users use Lutris and don't see it as outdated.

**Comment by strycore** [comment]
Debian 12 is shipping in less than a week and we now have support for Flatpak runners.

We can think about migrating to Gtk 4 pretty soon. 

As for Libadwaita, are the "it looks out of place" comments worth taking in consideration? In an age where you have tons of Electron apps each with their own style? Lutris is meant to be desktop agnostic. It should also look nice on the Steam Deck which uses Plasma.

I think that as long as Lutris ships by default with the Adwaita dark theme, I don't see an issue with it. Oh, and of course, I should be able to figure out how to use libadwaita from Python code. If there's no documentation and no other Python project using Libadwaita, I'm less likely to put myself in a torturous situation.

**Comment by danieljohnson2** [comment]
I'd ask if libawaita is really less theme-able than GTK4. It's not something Lutris absolutely needs, but it's clearly desirable.

If GTK4 has the edge here, then I'm not sure what libadwaita has to offer that we'd want to switch to it.

**Comment by CapeCrusader321** [comment]
Any Updates On This ??, Lutris Codebase is ancient , the devs should take steps to upgrade lutris to gtk4 if not libadwaita.

**Comment by strycore** [comment]
Debian 12 got released last week, geez...

We're releasing 0.5.14 using Gtk 3. I don't think there will be a 0.5.15. 0.6.0 will be Gtk 4 based.

**Comment by CapeCrusader321** [comment]
> Debian 12 got released last week, geez...
> 
> We're releasing 0.5.14 using Gtk 3. I don't think there will be a 0.5.15. 0.6.0 will be Gtk 4 based.

Thats a very good update, waiting for lutris 0.6 🎉

**Comment by cmpunches** [comment]
No, libadwaita is an anti-choice abomination that harms users.  It should be categorically rejected by any project that wants to see users use their own themes.  

The Gnome folks are pretending to be Apple, and Apple doesn't play well with open source.

**Comment by LeviLeal** [comment]
Not everyone loves libadwaita. Besides, it looks way more inconsistent with other DEs like plasma, cinamon or xfce, since its more difficult to theme it.

**Comment by strycore** [comment]
The way I see it, there is so little documentation about Python specific Gtk 4 (and libadwaita), the process of porting our current codebase to it is quite daunting.

With that in mind, it might be a good opportunity for Lutris to draw a clear separation between the backend and the UI. We still have plans for a Godot UI (although, we're most likely to help with the Lutris plugin for OpenGamePadUI). We might also consider using Qt if porting to Gtk4 proves to be too challenging.

**Comment by Artek52** [comment]
what is the status on this? will there be an adwaita port? a qt port? a Godot UI? I dont understand

**Comment by strycore** [comment]
- Godot "port": it's not a port, it's an alternative UI and it already exists, it's called OpenGamepadUI. We will focus on bringing a tighter integration with Lutris but it already exists.
- Qt port: very likely since it looks like Pyside 6 is everything I ever wanted from a UI toolkit and its documentation is so hot and sexy and I want to touch it.
- Adwaita port: very unlikely due to the controversial nature of Adwaita coupled with a  complete lack of documentation and a general "fuck you" attitude regarding Python. Not to mention that Gtk apps crashs when using a high DPI Razer mouse.

**Comment by Artek52** [comment]
off topic but I never understood what is so controversial about libadwaita, I dont know about developing with it but all adwaita apps are really beautiful and easy to use, much more that qt apps.
And I am saying from kde plasma

**Comment by strycore** [comment]
I dont have strong opinions about Adwaita but the vibe I'm getting from Adwaita apps is that most of them are microscopic in scope and as such have a very minimalistic UI whereas Qt is often used to build complex apps. That might explain why Adwaita feels cleaner than Qt.

**Comment by LeviLeal** [comment]
Personally, i don't like the libadwaita style, it looks like a mobile interface, it doesn't fit well with a powerfull tool like Lutris.

**Comment by cameronbosch** [comment]
> Personally, i don't like the libadwaita style, it looks like a mobile interface, it doesn't fit well with a powerfull tool like Lutris.

Then the only other option would be to rewrite everything in another toolkit like Qt. GTK 4 has become so entwined with LibAdwaita that they may as well be one.

Unless Lutris adopts LibAdwaita, I'd say they'd be better off using Qt or something other than GTK. Because GTK 3 isn't going to last forever.

**Comment by BrainBlasted** [comment]
> Adwaita port: very unlikely due to the controversial nature of Adwaita coupled with a complete lack of documentation and a general "fuck you" attitude regarding Python. Not to mention that Gtk apps crashs when using a high DPI Razer mouse.

Regarding the lack of documentation, what would you like to see? We try very hard to document the new APIs.

> Then the only other option would be to rewrite everything in another toolkit like Qt. GTK 4 has become so entwined with LibAdwaita that they may as well be one.

No, not really. GTK4 functions just fine as an independent toolkit - you just don't have libadwaita's widgets and styling.

**Comment by strycore** [comment]
> Regarding the lack of documentation, what would you like to see? We try very hard to document the new APIs.

That is surprisingly easy to answer: PySide has extremely good documentation. If Python + Gtk ever managed to get that level of quality, I would reconsider a lot of things.

There are examples for every thing, there are articles on specific topics, there are more examples than anyone could want. This is exactly what I've wanted for Gtk.

**Comment by Artek52** [comment]
> Personally, i don't like the libadwaita style, it looks like a mobile interface, it doesn't fit well with a powerfull tool like Lutris.

There is this little program called bottles. It has a similar scope to lutris and it uses libadwaita. It looks and works great. I think its just getting ridiculous. People hate on gtk because its associated with gnome. Not because they have actual complaints that make sense about it.

**Comment by strycore** [comment]
One of the reasons I'm holding back on migrating to Gtk 4 is because one of the main developers of Bottles told me how painful the process was.

**Comment by danieljohnson2** [comment]
I've looked at GTK 4, and it's definitely going to be a challenge to migrate there, but I am pretty sure QT would be harder still.

But the GTK documentation is pretty weak honestly, even if you are using C, and I've had to read the GTK source to figure things out at times. So I can understand why @strycore is attracted to QT, even if it'd be hard to get there.

I think both of these things have a common cause- GTK just *hates* compatibility. A lot of the changes I'm seeing in GTK 4 are pretty pointless, but they make it hard to migrate, and they mean the docs have needed to be updated, and GTK 3 web resources apply a lot less than they should to GTK 4.

And I'm sympathetic to the "let's do this *right*" attitude. I like GNOME more than KDE because of it; but it's clear to me they've gone too far. It's self-destructive.

**Comment by strycore** [comment]
What makes you think porting Lutris to Qt would be hard?

**Comment by strycore** [comment]
It's probably going to be long and tedious but I believe that if Qt has solid documentation, the process should be fairly straightforward

**Comment by danieljohnson2** [comment]
There's one big question for me- is it possible to build a program that part GTK, part QT?

If it is, a long (and tedious) conversion process is a lot more practical.

**Comment by strycore** [comment]
No it's not possible. Both Gtk and Qt implement their own event loop so you can't mix those 2 together.
We can port the code going from base functionality to more specific ones, starting with application.py and lutriswindow.py + including all non gtk modules directly from the start.

We have to make it very clear that we're not going to maintain 2 different widget toolkits. Once the Qt port is done, everything Gtk related needs to be abandoned. 
Nothing more ridiculous than software with a gtk and a qt version (looking at you Audacious and Transmission)

**Comment by strycore** [comment]
I like Audacious and Transmission too but I don't believe any community run open source project has the resources to maintain 2 GUI toolkit implementations. 
Not only that but it's also confusing for users as it's hard to know which one is the better one. There's very little chance that both implementations are exactly on par feature wise and quality wise.

**Comment by danieljohnson2** [comment]
Well, switching between GTK and QT wouldn't help us; it sounds like we're going to have a big chunky conversion that has to be done, tested, and released all together. That's going to be trouble.

The best thing we can do is get as much stuff out of GTK's icy grip before taking that plunge.

I was planning to work on that soon anyway; get the GObject out of non-UI classes. Mind if I just do it, and don't pitch you a PR for review? Having a PR like this out there is going to be very conflicty.

Also, if you like the idea of PR #5442, I think I can go further with it and pull searching code out of LutrisWindow and into this new GameSearch class.

I get the sense you'd like to have a more elaborate UI than GTK wants to support. Qt may be the answer to that, but I liked the GNOMEy UI, and I will miss my precious header bars. Ah well.

**Comment by axel358** [comment]
I don't see how qt widgets offers any advantage over GTK 4, and don't see anything that can't be done in GTK 4 that can on gtk3 or qt, being qt widgets the most basic of all feature wise. To me it seems more troublesome to port between toolkits than porting to GTK 4.

**Comment by danieljohnson2** [comment]
Well, I'm not too familiar with Qt; maybe it's crap, but GTK is pretty targeted on GNOMEy UI, and I think we'll have trouble realizing some of @strycore's ideas.

I mean, if it were up to me I'd go to GTK 4 instead, but I'm not the boss of Lutris.

**Comment by An-Eagle** [comment]
To be fair Qt looks horrible on gnome, while GTK, especially 4, is quite easy to theme on other desktop environments/WMs

**Comment by Artek52** [comment]
I agree and I think it is something to keep in mind. Gtk looks alright with kde and pretty good even on the new cosmic-epoch by sytem76. Qt looks alien on anything but kde.

**Comment by strycore** [comment]
> I don't see how qt widgets offers any advantage over GTK 4,

I see a few things where PySide is a better option than Gtk 4:

- Proper documentation for Python (even if a lot of it is ported from the C++ docs, I haven't noticed any errors so far)
- Lots of code examples (again, automatically ported from C++ but the ones I've tried worked)
- Code completion in editors
- First class support for virtual environment
- First class support for bundling an application as a single executable
- Better support for changing the look and feel of an application

**Comment by strycore** [comment]
> Mind if I just do it, and don't pitch you a PR for review?

No matter which toolkit we end up choosing, this is always welcome

**Comment by danieljohnson2** [comment]
> No matter which toolkit we end up choosing, this is always welcome

I'll take that as a 'yes'!

**Comment by strycore** [comment]
yes

**Comment by mfreeman72** [comment]
Please, PLEASE, *PLEASE*, **PLEASE**, ***PLEEEEEAAASSE*** don't use libadwaita. I purposely avoid anything using that horrifyingly ugly, lifeless, unthemeable mess. Everything using libadwaita looks nasty and out of place in anything but Gnome, where it's "everyone must use crapdwaita or nothing". If I install a program, and all I see when I launch it is Adwaita, and not my beautiful theme that I painstakingly crafted to make my desktop a work of art, it's immediately uninstalled and blacklisted. No thank you!

**Comment by An-Eagle** [comment]
> Please, PLEASE, _PLEASE_, **PLEASE**, _**PLEEEEEAAASSE**_ don't use libadwaita. I purposely avoid anything using that horrifyingly ugly, lifeless, unthemeable mess. Everything using libadwaita looks nasty and out of place in anything but Gnome, where it's "everyone must use crapdwaita or nothing". If I install a program, and all I see when I launch it is Adwaita, and not my beautiful theme that I painstakingly crafted to make my desktop a work of art, it's immediately uninstalled and blacklisted. No thank you!

you do realise that libadwaita can be pretty well themed anyways, and qt is the exact same, it looks horrible with anything but kde, so your argument makes no sense

**Comment by strycore** [comment]
As someone who uses Qt apps on Gnome and Gtk apps on KDE, I have no idea what yall are on about.

At least post some screenshots on how it's "out of place", otherwise it's just baseless.

**Comment by mfreeman72** [comment]
> you do realise that libadwaita can be pretty well themed anyways, and qt is the exact same, it looks horrible with anything but kde, so your argument makes no sense

Could you please explain how to easily get libadwaita to match my preferred GTK themes, if it can be "pretty well themed"? The only ways I can find are difficult and hackish.

Also, I've never had any problems with QT/KDE apps looking very close to my GTK theme. Yeah, it's not exactly perfect, but close enough that they don't look out of place. And it all happens pretty much automatically when I change my GTK theme. But I've never found an easy/automatic way to make libadwaita look like anything but the default libadwaita, except maybe changing the color of the lipstick on the ugly pig.

**Comment by An-Eagle** [comment]
> > you do realise that libadwaita can be pretty well themed anyways, and qt is the exact same, it looks horrible with anything but kde, so your argument makes no sense
> 
> Could you please explain how to easily get libadwaita to match my preferred GTK themes, if it can be "pretty well themed"? The only ways I can find are difficult and hackish, and nowhere close to being an easy, automatic process, where when I change my GTK theme, libadwaita automatically follows.
> 
> Also, I've never had any problems with QT/KDE apps looking very close to my GTK theme. Yeah, it's not exactly perfect, but close enough that they don't look out of place. And it all happens pretty much automatically when I change my GTK theme. But I've never found an easy/automatic way to make libadwaita look like anything but the default libadwaita, except maybe changing the color of the lipstick on the ugly pig.

the theme is stored in .config/gtk-4.0, not in dbus

**Comment by mfreeman72** [comment]
> the theme is stored in .config/gtk-4.0, not in dbus

Who said anything about dbus?

**Comment by probonopd** [comment]
Please don't. [libadwaita doesn't work well](https://www.youtube.com/watch?v=WFqJA1q-IKY) outside of the Gnome desktop.  Switching to libadwaita would significantly reduce the potential user base for this application, essentially limiting it to the fraction of users who happen to use Gnome.

**Comment by QmwJlHuSg9pa** [comment]
> Please don't. [libadwaita doesn't work well](https://www.youtube.com/watch?v=WFqJA1q-IKY) outside of the Gnome desktop. Switching to libadwaita would significantly reduce the potential user base for this application, essentially limiting it to the fraction of users who happen to use Gnome.

The last time I tried using Gradience (a recently discontinued third-party app intended to assist hacking around libAdwaita theming limitations), I attempted to apply a linear-gradient to the libAdwaita "headerbar", only to find that my desired gradient was forcibly overridden by a solid RGB red (#FF0000) -- to me, this seems a rather intentional attempt at sabotaging viability of customisation on part of the GNOME shell implementers.

Never mind that attempts at hacking around libAdwaita (which boils down to custom CSS rules) inadvertently break almost every major GNOME release... Currently, my "Files" sidebar has lost it custom background colour, rendering the text within almost illegible.

And from my experiences on the GitLab issue trackers, libAdwaita and some GNOME app maintainers seem actively hostile to the notion of user choice. Whether it be sidebar functionality (silent removal of a comment I made regarding user-choice sidebar width in the Files app, vs. pre-calculated fixed-width); or implementing additional color themes within a GNOME Circle text editor, e.g. Solarized (can't do it, won't do it, stop responding -- we only do Dark, Light or Sepia(???) -- and no, CSS rules can't override these built-in themes...).

I believe libAdwaita is narrow-minded in its mission, to a fault. If you merely wanting to customise any aspect of the experience beyond libAdwaita's impenetrable "vision", you're bound to be told "you're using it wrong" and inevitably silenced from official channels.

**Comment by v3xed-v0id** [comment]
libadwaita is not a good idea and its fully gnome-centric, its not very good outside of gnome, i recommend using qt rather than libadwaita if a toolkit switch was gonna occur, thanks for reading.

**Comment by v3xed-v0id** [comment]
i use plasma and a gnome/libadwaita switch would cause me to abandon lutris over accessibility issues. dont switch to libadwaita please. no point in libadwaita.

**Comment by v3xed-v0id** [comment]
> Please don't. [libadwaita doesn't work well](https://www.youtube.com/watch?v=WFqJA1q-IKY) outside of the Gnome desktop. Switching to libadwaita would significantly reduce the potential user base for this application, essentially limiting it to the fraction of users who happen to use Gnome.

one of the few times i agree with probonopd

**Comment by mfreeman72** [comment]
> the theme is stored in .config/gtk-4.0, not in dbus

Also, when I do copy my theme over to the .config/gtk-4.0 folder (which I have to manually do, there's nothing automatic about it like in every other UI available, including QT), it *tries* to retheme libadwaita windows to my theme, but everything's out of whack. Sidebars no longer properly line up with the rest of the window, many GUI elements don't show at all, and in some cases, there are big, ugly gray circles around my window control widgets. I just looks ugly. So no, I consider libadwaita unthemable on my system. It's a terrible excuse for a GUI toolkit.

**Comment by v3xed-v0id** [comment]
literally every other desktop other than gnome implements SSD and app theming and other standard accessibility solutions for issues, gnome is the only one that lacks these and so does their libadwaita toolkit (which in itself only really works "well" on gnome and not other desktops like plasma or cinnamon)

**Comment by v3xed-v0id** [comment]
> > the theme is stored in .config/gtk-4.0, not in dbus
> 
> Also, when I do copy my theme over to the .config/gtk-4.0 folder (which I have to manually do, there's nothing automatic about it like in every other UI available, including QT), it _tries_ to retheme libadwaita windows to my theme, but everything's out of whack. Sidebars no longer properly line up with the rest of the window, many GUI elements don't show at all, and in some cases, there are big, ugly gray circles around my window control widgets. I just looks ugly. So no, I consider libadwaita unthemable on my system. It's a terrible excuse for a GUI toolkit.

gnome devs explicitly stated they dont like app theming and are trying to strongarm other devs into not supporting theming, libadwaita is one of the ways they are forcing devs from not supporting theming, they eventually wanna fully remove it, this is one of their websites: https://stopthemingmy.app

**Comment by v3xed-v0id** [comment]
gnome devs think that app theming is bad even though theming is necessary for accessibility, gnome devs are so blinded by their overly-opinionated ideas of a desktop they dont realize they are hurting accessibility

they claim to be accessibility champions but its all just virtue signalling, they dont actually care about accessibility.

**Comment by tockudex** [comment]
@v3xed-v0id your point is understandable and fair, but I don't see why it would be helpful to publish more than one comment about it.

**Comment by v3xed-v0id** [comment]
> @v3xed-v0id your point is understandable and fair, but I don't see why it would be helpful to publish more than one comment about it.

mb ur right, i drank too much caffeine today, im gonna go take a break, cya

**Comment by EuroBlue** [comment]
How about to do the sane thing and port to GTK 4 but not to libadwaita.

**Comment by danieljohnson2** [comment]
> How about to do the sane thing and port to GTK 4 but not to libadwaita.

That's what I favor, but it's hard to port to GTK4 because GNOME hates compatibility. There's a lot of change for the sake of change in there.

**Comment by v3xed-v0id** [comment]
> > How about to do the sane thing and port to GTK 4 but not to libadwaita.
> 
> That's what I favor, but it's hard to port to GTK4 because GNOME hates compatibility. There's a lot of change for the sake of change in there.

it could also be considered that lutris could simply just stay on GTK3 and not update to GTK4 like many desktop environments have decided to do (Cinnamon, XFCE, MATE, etc), its perfectly valid to not update the toolkit at all!

**Comment by EuroBlue** [comment]
> > > How about to do the sane thing and port to GTK 4 but not to libadwaita.
> > 
> > 
> > That's what I favor, but it's hard to port to GTK4 because GNOME hates compatibility. There's a lot of change for the sake of change in there.
> 
> it could also be considered that lutris could simply just stay on GTK3 and not update to GTK4 like many desktop environments have decided to do (Cinnamon, XFCE, MATE, etc), its perfectly valid to not update the toolkit at all!

GTK4 is superior tech. They are not some default behaviour settings but bring a ton of new functionality for example for design layouting, performance and accessibility. And I can guarantee you that Cinnamon with time will change to GTK4 they just have more important stuff to do.

**Comment by EuroBlue** [comment]
For any GNOME fanatics there is [Cartridges](https://apps.gnome.org/Cartridges/). But porting to libadwaita right now is saying: "This is a program made explicitly for GNOME" because that is what libadwaita is. I love the GNOME HIG, Design and etc but libadwaita's complete artificial blocks on user-themeing, window decorations, and custom CSS make it not suitable for cross-desktop programs like lutris. If libadwaita changes (which i hope it does) then it is possible.

**Comment by LeviLeal** [comment]
Lutris being ported to QT will make it look better on Steam Deck. We already have alternatives made for GTK, as bottle or cartridges

**Comment by EuroBlue** [comment]
@LeviLeal neither bottle or cartridges are a GTK alternatives! Cartridges are libadwaita (I explained why libadwaita is bad in my previous comment). Bottles are now libadwaita but they'll change to libcosmic and bottles doesn't even has the needed functionality to be considered. GTK is the most used linux Widget Toolkit. Only KDE and LXQT use QT. And porting to QT means that it will look native only in the desktop mode. There is an alternative to all this. It is for lutris to split up into a server client model where there can be different UIs that use the same backend. That would in my opinion be the best possible outcome, but it is just to much work for me to even have the audacity to seriously propose it now since there are many other more important things that need to be done.

On the other hand I have a question for @strycore . Wouldn't it be useful for Lutris to adapt this server client model for example as part of your work with playtron so that playtrons gamepadui can make a native for them looking GUI but still use all the improvements of lutris?

**Comment by primelemma** [comment]
Libadwaita app looks alian on other desktops, this same going to Happen with qt app, qt only look perfect in kde plasma.

Argument should be which is easy for development and management.
porting to gtk3 to gtk4 is easier than qt, but management of qt app is easy i think because of stable APIs.
but qt apps are always going to look ugly like every kde apps.

**Comment by hyperballicz** [comment]
> but qt apps are always going to look ugly like every kde apps.

This is strongly opinionated, not a fate.
For me, all LibAdwaita apps looks horrendous, they look like mobile apps and are too annoying to configure, too much spacing and little information density. And looking at lutris how it is right now, besides the use of CSD, it fits way more to the KDE design language than LibAdwaita.
EasyEffects recently moved to QT, and i find it way prettier than the older libAdwaita version [https://wwmm.github.io/easyeffects/images/easyeffects-light-screenshot-3.png](https://wwmm.github.io/easyeffects/images/easyeffects-light-screenshot-3.png)
And if you just want a libadwaita Wine prefix management tool, you're already have bottles

**Comment by hyperballicz** [comment]
> [@LeviLeal](https://github.com/LeviLeal) neither bottle or cartridges are a GTK alternatives! Cartridges are libadwaita (I explained why libadwaita is bad in my previous comment). Bottles are now libadwaita but they'll change to libcosmic and bottles doesn't even has the needed functionality to be considered. GTK is the most used linux Widget Toolkit. Only KDE and LXQT use QT. And porting to QT means that it will look native only in the desktop mode. There is an alternative to all this. It is for lutris to split up into a server client model where there can be different UIs that use the same backend. That would in my opinion be the best possible outcome, but it is just to much work for me to even have the audacity to seriously propose it now since there are many other more important things that need to be done.
> 
> On the other hand I have a question for [@strycore](https://github.com/strycore) . Wouldn't it be useful for Lutris to adapt this server client model for example as part of your work with playtron so that playtrons gamepadui can make a native for them looking GUI but still use all the improvements of lutris?

Bottles will not drop the libadwaita version

**Comment by woj-tek** [comment]
> Libadwaita app looks alian on other desktops, this same going to Happen with qt app, qt only look perfect in kde plasma.

> This is strongly opinionated, not a fate.

my 5c:

IMHO while Qt apps is hit-or miss, they usually tend to at least try to look native-ish on other OSs (and they usually do a decent job).

Anything with libadwaita is just a soreeye on anything except for GNOME and is a pain to use most of the time…

**Comment by strycore** [comment]
> Wouldn't it be useful for Lutris to adapt this server client model for example as part of your work with playtron so that playtrons gamepadui can make a native for them looking GUI but still use all the improvements of lutris?

OpenGamepadUI isn't part of Playtron and there is a Lutris plugin for it already (which is pretty basic, it needs a lot of work). But I agree, it's a good idea to split in a server / client model and I totally think it's doable with the development tools we have now.

**Comment by woj-tek** [comment]
> But I agree, it's a good idea to split in a server / client model and I totally think it's doable with the development tools we have now.

hmm… client-server seems very sensible IMHO. Though I thought you kinda discarded this idea? (https://github.com/lutris/lutris/issues/3886#issuecomment-2077660386 & https://github.com/lutris/lutris/issues/3886#issuecomment-2077692404). Or only the notion of having (and self-maintaining) different client implementations within the project?

**Comment by strycore** [comment]
this was more about having to maintain 2 separate UI toolkits whitin the project. Having a client server model would allow to migrate from 1 UI to another more easily

**Comment by jedace07** [comment]
Is there any definitive plan to move off GTK 3 for Lutris 0.6?

**Comment by danieljohnson2** [comment]
Pretty much. I've got [this here branch](https://github.com/lutris/lutris/tree/dj/gtk4) that ~~Claude~~I have put together to port to GTK 4 and I am debugging it; it's gotten pretty far along. Feel free to give it a try!

You'd have to ask @strycore when this will actually happen; I think there are meant to be two more 0.5.x release to come first.

**Comment by strycore** [comment]
> I think there are meant to be two more 0.5.x release to come first.

Ideally, just one? We have to test and release 0.5.23 but then we can test 0.6.0 with the gtk4 branch

**Comment by MorsMortium** [comment]
Not to bring back any arguments about CSD, just mentioning for anyone that does not want it, that I have created a new GTK-NoCSD for all GTK versions (GTK3, GTK4, libhandy, libadwaita) and with  support in case something is broken (https://codeberg.org/MorsMortium/GTK-NoCSD) I will test GTK4 Lutris with it as soon as it gets to Arch/Artix repos
